src/components/message-actions.js

```diff
@@ -44,8 +44,25 @@ class MessageActions extends CustomElement {
         this.chatview.onMessageRetractButtonClicked(this.model);
     }
 
+    async onBanUserButtonClicked (ev) {
+        ev.preventDefault();
+        const reason = await api.prompt(__('Reason'), __('Type a reason for the ban'));
+        this.chatview.onBanUser(this.model, reason);
```

Comment:
Did you test this code?

You're calling `onBanUser` on the view, but you created `onBanUser` on the model (which is IMO the right place). So you'll get an AttributeError when this code runs.

Every view has a `model` attribute which references the model it applies to, so you could call `this.chatview.model.onBanUser(this.model, reason);` (in case it's not clear, `this.chatview.model` is the `ChatBox` instance and `this.model` is the `Message` instance).

Also, for an event handler, I think using the `on` prefix is fine, but it doesn't make sense to me for general methods. So I think `onBanUser` should be renamed to just `banUser`.

That said, it's not clear to me that we even need a `banUser` method, I think you could probably just call `this.chatview.model.sendAffiliationIQ` from here.